### ISSUE TITLE
Skip or Exclude not working on play-tests

### ISSUE THREAD

**Issue by AmirTugi** [storybookjs/test-runner#424]
**Describe the bug**
I've added the `skip` tag (also tried with the `exclude` tag) to skip currently failing a11y tests, and I still see that some (or maybe all) the `play-test`s are running, regardless of the "skip" tag I added to them.

I noticed that the `smoke-test`s are skipped and that works fine. It's just the `play-test`.

**Expected behavior**
The `play-test` should also be skipped.

**Screenshots**

<img width="1113" alt="image" src="https://github.com/storybookjs/test-runner/assets/8065975/91924fa9-d4af-45a0-9912-a4d1eb7d0f26">

<img width="484" alt="image" src="https://github.com/storybookjs/test-runner/assets/8065975/19a8669d-2277-476f-bc24-4cfbf7f1254a">

<img width="307" alt="image" src="https://github.com/storybookjs/test-runner/assets/8065975/2e398bfe-52a4-4708-957f-fb20040285c3">

**System**

Storybook Environment Info:

  System:
    OS: macOS 14.2.1
    CPU: (12) x64 Apple M2 Max
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 20.10.0 - ~/.volta/tools/image/node/20.10.0/bin/node
    Yarn: 1.22.19 - ~/.volta/tools/image/yarn/1.22.19/bin/yarn <----- active
    npm: 10.2.3 - ~/.volta/tools/image/node/20.10.0/bin/npm
  Browsers:
    Chrome: 120.0.6099.234
    Safari: 17.2.1

**Comment by AmirTugi** [comment]
I managed to find a workaround for now, and to add `tags: ['a11y-skip']` directly to the story, and not in the component meta.
Seems like I was wrong about the `async` aspect.
It seems like that when `tags` is in the component-level - it works only for `smoke-test`s, and when it's for a specific story - it's also for its `play-test`.

**Comment by renatewr** [comment]
We had the exact same issue, and have spent hours debugging.

The link to the erroring story/play test is working, but running `test-storybook` the "play test" logged out "message: null".

I can confirm that this workaround works - the story is then excluded.

However, I would like to figure out _why_ it fails, but I cannot get any debug information, since the built story is working fine.
I _suspect_ that the reason is that the story is actually loading a video player.

**Comment by ahayes91** [comment]
Had some trouble with tags myself today, and tracked it down to the cache - https://github.com/storybookjs/test-runner/blob/fd75838863b8dcac8f09ba8c17cc777e7a16a679/src/util/getTestRunnerConfig.test.ts#L52

Once I started using `--no-cache` flag, changes to the `tags` attributes in test-runner.js file were being picked up correctly. 
https://storybook.js.org/docs/writing-tests/integrations/test-runner#cli-options
